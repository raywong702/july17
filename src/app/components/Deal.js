import React from 'react';

class Deal extends React.Component {
  render() {
    return (
      <div className="deal" style={
        { color: this.props.price > 10000 ? 'green' : 'red' }
      }>
        <h3>{this.props.title}</h3>
        <div>{this.props.cause.name}</div>
        <p>${this.props.price / 100}</p>
      </div>
    );
  }
}

export default Deal;
