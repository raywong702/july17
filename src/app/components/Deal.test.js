import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import React from 'react';
import { Deal } from './Deal';
import { render } from 'enzyme';

describe('Deal', () => {
  it('works', () => {
    const wrapper = render(<Deal />);

    expect(wrapper).toMatchSnapshot();
  });
});
