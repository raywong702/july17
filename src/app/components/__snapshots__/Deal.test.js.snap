// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deal works 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "deal",
      "style": "color:red",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [],
        "name": "h2",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
